tests/phpunit/integration/Core/Util/Entity_FactoryTest.php

```diff
@@ -0,0 +1,765 @@
+				),
+				// TODO: This should be 'https://example.com/blog/customposttype/page/3/'.
+				new Entity(
+					'https://example.com/blog/customposttype/',
+					array(
+						'type'  => 'post_type',
+						'title' => 'Archives: Custom Post Type',
+					)
+				),
+			),
+			'taxonomy archives'            => array(
+				array(
+					'customtaxonomy' => 'coffee',
+				),
+				new Entity(
+					'https://example.com/blog/customtaxonomy/coffee/',
+					array(
+						'type'  => 'term',
+						'title' => 'Custom Taxonomy: Coffee',
+						'id'    => 0, // Filled via title lookup.
+					)
+				),
+			),
+			'taxonomy archives, page 3'    => array(
+				array(
+					'customtaxonomy' => 'coffee',
+					'paged'          => '3',
+				),
+				// TODO: This should be 'https://example.com/blog/customtaxonomy/coffee/page/3/'.
+				new Entity(
+					'https://example.com/blog/customtaxonomy/coffee/',
+					array(
+						'type'  => 'term',
+						'title' => 'Custom Taxonomy: Coffee',
+						'id'    => 0, // Filled via title lookup.
+					)
+				),
+			),
+		);
+	}
+
+	protected function assertEntity( Entity $expected, $actual ) {
+		$this->assertInstanceOf( Entity::class, $actual );
+		$this->assertSame( $expected->get_url(), $actual->get_url(), "Failed asserting that entity URL {$actual->get_url()} is {$expected->get_url()}." );
+		$this->assertSame( $expected->get_type(), $actual->get_type(), "Failed asserting that entity type {$actual->get_type()} is {$expected->get_type()}." );
+		$this->assertSame( $expected->get_title(), $actual->get_title(), "Failed asserting that entity title {$actual->get_title()} is {$expected->get_title()}." );
+		$this->assertSame( $expected->get_id(), $actual->get_id(), "Failed asserting that entity ID {$actual->get_id()} is {$expected->get_id()}." );
+	}
+
+	private function set_and_authorize_current_user( \WP_User $user ) {
+		wp_set_current_user( $user->ID );
+		// The only way to change the current user in bootstrapped Site Kit class instances (e.g. Permissions)
+		do_action( 'wp_login', $user->user_login, $user );
+
+		// Fake setup and authentication for access to dashboard.
+		$this->fake_proxy_site_connection();
+		remove_all_filters( 'googlesitekit_setup_complete' );
+		$authentication = new Authentication( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$authentication->verification()->set( true );
+		$authentication->get_oauth_client()->set_access_token( 'test-access-token', HOUR_IN_SECONDS );
+	}
+
+	/**
+	 * Extracts the title of a WordPress entity (e.g. a post, term or user) from a Site Kit entity.
+	 *
+	 * Site Kit entities are based on URLs and therefore use the real title for those URLs, which often have a prefix
+	 * separated by the real entity title with ': '. This method strips out the prefix.
+	 *
+	 * @param string $title Site Kit entity title (potentially including ': ').
+	 * @return string Actual WordPress entity title.
+	 */
+	private function extract_wp_entity_title_from_url_entity_title( $title ) {
```

Comment:
This method becomes unnecessary as well once we no longer need to mutate the expected entity in the test 🙂